npe2/manifest/schema.py

```diff
@@ -105,6 +135,20 @@ def description(self) -> Optional[str]:
     def author(self) -> Optional[str]:
         return self.package_metadata.author if self.package_metadata else None
 
+    @root_validator
+    def _check_engine_version(cls, values: dict) -> dict:
+        declared_version = semver.VersionInfo.parse(values.get("engine", ""))
+        current_version = semver.VersionInfo.parse(ENGINE_VERSION)
```

Comment:
would prefer not to bring on a dependency just for this.  is it possible?

Reply:
(it's our own version string... so it's not like we need to be ready for anything beyond `version.split('.')` ... if someone puts some ridiculous unrecognized engine version in their manifest that fails to parse, i think that's a reasonable exclusion criterion)

Reply:
I agree this is a bad dependency (semver's about to release 3.0 and it'll break things).

It gives us parsing and comparison which are easy to do custom.

I'm a little busy so I'd like to just make an issue to replace the dependency later so we can go ahead and merge this. But it's not a lot of work..